File: php/threexplusone.php
#!/usr/bin/php
<?php
function threexplusone($x) 
{
	if ($x == 1) {
		return [1];
	} else if ($x % 2 == 1) {
		return array_merge(threexplusone(3 * $x + 1), [$x]);
	} else {
		return array_merge(threexplusone($x / 2), [$x]);
	}
}

print_r(array_map('threexplusone', range(1, 10000)));

File: python/threexplusone.py
#! /usr/bin/python3
def collatz(i):
    ret = [i]
    while i != 1:
        i = 3 * i + 1 if i & 1 else i // 2
        ret.append(i)
    return ret

for i in range(1,10001):
    print(collatz(i))